Sim on range2

joshuef opened the pull request.

grumbach commented on sn_interface/src/network_knowledge/node_state/mod.rs:
```diff
@@ -111,16 +111,19 @@ impl NodeState {
-                    .values()
-                    .find(|n| n.peer().addr() == self.peer().addr())
-                {
+                } else if let Some(existing_node) = members.values().find(|n| {
+                    // reject if we find a node that is marked as currently 'Joined' AND has same socket.
+                    // (This check is needed to allow nodes that have been Relocated to rejoin this section
+                    // if they've been sent to our section again)
+                    n.peer().addr() == self.peer().addr()
+                        && matches!(n.state, MembershipState::Joined)
```
`members` above come from the call to `self.section_members` which already filters for `MembershipState::Joined`

joshuef replied:
orly 🤔

grumbach reviewed (approved):
LGTM

joshuef committed:
feat(networking): initial range based gets implementation

joshuef committed:
chore(networking): rename sort_peers fns to clarify the accepted limit arg

joshuef committed:
chore(networking): remove REPLICATION_PEER_COUNT

joshuef committed:
fix(networking): reset raw spend get to be Majority

joshuef committed:
fix(networking): return true for exceeded quorum for non sensitive data

joshuef committed:
fix(networking): use cloest_node_buffer_zone when selecting nodes for range replication

joshuef committed:
fix(networking): use libp2p for register retrieval

merge the results after we see split record.
Use Quorum::Majority at least (instead of Quorum::One)

joshuef committed:
chore(protocol): move get_type_from_record to protocol

joshuef committed:
fix(networking): remove self from get_peers_excluding_self

joshuef committed:
chore: small refactors and doc tweaks

joshuef committed:
fix(networking): remove custom REPLICATION_FACTOR

this indirectly affects how many close nodes returned and could stymie
the GetRange effectiveness

joshuef committed:
ci: normalize send amount for tests

joshuef committed:
ci: fixes

joshuef committed:
fix(client): retry faucet loading if genesis is spent

joshuef committed:
fix(faucet): loop if initial faucet balance 0 but genesis spent

joshuef committed:
fix(networking): ensure we update getrange when we can

ie, even if query times out if we have peers, we update

seems like we timeout more due to high K_VALUE

joshuef committed:
fix(networking): redial peers on intermittent errors

previously we just dropped any attempt at comms there

joshuef committed:
fix(networking): clarify faucet error and update test

joshuef committed:
ci: clarify duplicate genesis tests in merge.yml

joshuef committed:
fix(networking): local network dont force cleanup bootstrap

We were cleaning up bootstrap even if not full just because of being local

joshuef committed:
ci: split out cashnote check for clarity

joshuef committed:
test(networking): clarify that _at least_ all in range keys should be fetched

allow more to be fetched if needed due to distance variation

joshuef committed:
ci: assert faucet funded before upload

joshuef committed:
chore(net): remove clone post-rebase

joshuef committed:
ci: wait for faucet completion in dist test

joshuef committed:
chore(networking): small fixes comment updates

joshuef committed:
chore: post-rebase update get register call

joshuef committed:
fix: prevent recursive bootstrap quereis on peer addition

joshuef committed:
fix(nodeman): allow not found errors during cleanup

joshuef committed:
test(networking): initial dabble in testnet / sim comparison

joshuef committed:
hmm

joshuef committed:
sanityQQQQ

joshuef committed:
hm

joshuef committed:
bar

joshuef committed:
okay

joshuef committed:
hmm

joshuef committed:
mosaneqqq

joshuef committed:
broken
